code/React/BootstrapComponent.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace SilverStripe\Admin\React;
+
+use SilverStripe\Core\Convert;
+
+trait BootstrapComponent
+{
+
+    private static $casting = [
+        'AttributesHTML' => 'HTMLFragment'
+    ];
+
+    private $attributes = [];
+
+    protected $extraClasses = [];
+
+    public function forTemplate()
+    {
+        $return = $this->renderWith($this->getTemplates());
```

Comment:
might be nice to throw somewhere if we aren't on viewable data, just in case someone expects that to work